table/server/common/src/main/java/alluxio/table/common/transform/TransformDefinition.java

```diff
@@ -26,18 +26,21 @@
 public class TransformDefinition {
   private final String mDefinition;
   private final List<TransformAction> mActions;
+  private final Properties mProperties;
 
   /**
    * The user-provided definition is normalized by:
    * 1. trimming whitespaces and semicolon from the beginning and end;
    * 2. normalize to lower case.
-   *
-   * @param definition the string definition
+   *  @param definition the string definition
    * @param actions the list of actions
+   * @param properties the list of properties extracted from definition
    */
-  private TransformDefinition(String definition, List<TransformAction> actions) {
+  private TransformDefinition(String definition, List<TransformAction> actions,
+                              Properties properties) {
```

Comment:
the properties and the definition represent the same transformation, right? Then, can we just pass in one of them, and derive the other?

Reply:
We can. But I would like to do that later. This constructor is also not public and the way this constructor gets called is by taking a definition and getting the properties.

Reply:
Does doing it later mean refactoring it later? I am fine with that. If so, can you leave a TODO?